src/matchbox/index.py

```diff
@@ -128,17 +129,17 @@ class MatchIndex:
     +-----------+---------------------+
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
         """Initialize the index."""
-        self.mismatch_unknown = set()
+        self.mismatch_unknown: Set = set()
```

Comment:
Consider generics: `MatchIndex[T]` for a `T` bound by `Hashable`.

Or do we want different types of entities in a single index?

Reply:
Could you check current use, if I haven't overdid something?